src/Microsoft.AspNetCore.Server.IISIntegration/Server/IO/AsyncIOOperation.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks.Sources;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration
+{
+    internal abstract class AsyncIOOperation: IValueTaskSource<int>, IValueTaskSource
+    {
+        private static readonly Action<object> CallbackCompleted = _ => { Debug.Assert(false, "Should not be invoked"); };
```

Comment:
Throw instead here?

Reply:
Nah, it's not worth throwing this in customer apps.